torchbenchmark/models/dcgan/__init__.py

```diff
@@ -0,0 +1,380 @@
+import numpy as np
+
+from pathlib import Path
+
+from ...util.model import BenchmarkModel
+from torchbenchmark.tasks import COMPUTER_VISION
+
+class DCGAN:
+ def __init__(self, bench):
+
+  # Set random seed for reproducibility
+  self.manualSeed = 999
+
+  random.seed(self.manualSeed)
+  torch.manual_seed(self.manualSeed)
+
+  # Number of workers for dataloader
+  self.workers = 2
+
+  # Batch size during training
+  self.batch_size = 32 #1024
+
+  # Spatial size of training images. All images will be resized to this
+  #   size using a transformer.
+  self.image_size = 64
+
+  # Number of channels in the training images. For color images this is 3
+  self.nc = 3
+
+  # Size of z latent vector (i.e. size of generator input)
+  self.nz = 100
+
+  # Size of feature maps in generator
+  self.ngf = 64
+
+  # Size of feature maps in discriminator
+  self.ndf = 64
+
+  # Number of batches to train with
+  self.num_train_batch = 1
+
+  # Number of training epochs
+  self.num_epochs = 5
+
+  # Learning rate for optimizers
+  self.lr = 0.0002
+
+  # Beta1 hyperparam for Adam optimizers
+  self.beta1 = 0.5
+
+  # Number of GPUs available. Use 0 for CPU mode.
+  self.ngpu = 1
+
+  self.device = bench.device
+
+# custom weights initialization called on netG and netD
+def weights_init(m):
+    classname = m.__class__.__name__
+    if classname.find('Conv') != -1:
+        nn.init.normal_(m.weight.data, 0.0, 0.02)
+    elif classname.find('BatchNorm') != -1:
+        nn.init.normal_(m.weight.data, 1.0, 0.02)
+        nn.init.constant_(m.bias.data, 0)
+
+class Generator(nn.Module):
+    def __init__(self, dcgan):
+        super(Generator, self).__init__()
+        self.ngpu = dcgan.ngpu
+        self.main = nn.Sequential(
+            # input is Z, going into a convolution
+            nn.ConvTranspose2d( dcgan.nz, dcgan.ngf * 8, 4, 1, 0, bias=False),
+            nn.BatchNorm2d(dcgan.ngf * 8),
+            nn.ReLU(True),
+            # state size. (dcgan.ngf*8) x 4 x 4
+            nn.ConvTranspose2d(dcgan.ngf * 8, dcgan.ngf * 4, 4, 2, 1, bias=False),
+            nn.BatchNorm2d(dcgan.ngf * 4),
+            nn.ReLU(True),
+            # state size. (dcgan.ngf*4) x 8 x 8
+            nn.ConvTranspose2d( dcgan.ngf * 4, dcgan.ngf * 2, 4, 2, 1, bias=False),
+            nn.BatchNorm2d(dcgan.ngf * 2),
+            nn.ReLU(True),
+            # state size. (dcgan.ngf*2) x 16 x 16
+            nn.ConvTranspose2d( dcgan.ngf * 2, dcgan.ngf, 4, 2, 1, bias=False),
+            nn.BatchNorm2d(dcgan.ngf),
+            nn.ReLU(True),
+            # state size. (dcgan.ngf) x 32 x 32
+            nn.ConvTranspose2d( dcgan.ngf, dcgan.nc, 4, 2, 1, bias=False),
+            nn.Tanh()
+            # state size. (dcgan.nc) x 64 x 64
+        )
+
+        self.jt = None
+        self.jitshape = None
+        self.debug_print = False
+
+    def forward(self, input):
+
+      if self.debug_print:
+        print(input.shape)
+
+      # Not enabled
```

Comment:
May I ask why disabling this code block?

Reply:
Its for dealing with dynamic shapes. I've removed it.